sound/soc/soc-core.c

```diff
@@ -1134,6 +1135,31 @@ static int soc_bind_dai_link(struct snd_soc_card *card,
 	}
 
 	soc_add_pcm_runtime(card, rtd);
+
+	/* if this is a virtual FE link, create runtime and set it as active */
```

Comment:
I don't know what 'active' runtime means. I would think it becomes 'active' when the switch is On, but maybe it's a different concept you are referring to.

Reply:
Oh, what I mean by active here is when the kcontrol is switched on and the dpcm_runtime_update() is called, it checks to see if playback or capture is active on the FE link. So I set the FE playback/capture active status by default.

Reply:
I see, but I wonder if this has negative implications related to suspend/resume? If there is nothing going on it's a bit odd to force a runtime to be active.

Reply:
Yes, thats a good point.

The problem is the runtime_update() method is oblivious to where it is being called from. It just looks for an active link to process the update. 

Maybe I can avoid this by activating the FE in the kcontrol IO handler just before calling runtime_update.